dags/common/pull_ftp.py

```diff
@@ -186,4 +186,4 @@ def trigger_file_processing(
 
 
 def _generate_id(publisher: str):
-    return datetime.utcnow().strftime(f"{publisher}_%Y-%m-%dT%H:%M:%S.%f")
+    return datetime.utcnow().strftime(f"{publisher}__%Y-%m-%dT%H:%M:%S.%f")
```

Comment:
why this change? it will affect all the ids from now on?

Reply:
because we have triggering the pull_ftp/fecth_api dags' tasks, we have ids generated with double dashes and then date, for example:
scheduled__2023-11-27T10:00:00+00:00
manual__2023-11-27T10:28:35.902469+00:00

Since the process dags' tasks we are triggering with a trigger, we are specifying the ids manually with _generate_id id function. To keep the names in consistent pattern allows me to select the run from DB and extract dates correctly

Reply:
@ErnestaP thanks, but still I cannot understand. Why instead of changing the id, we change just the query?

Reply:
The query is new. In order to get results we should have a template, for how the run_ids are named. To have it consistent with autogenerated ids (airflow run_ids) and our named run_ids (runs that we are triggering by the trigger tasks:
https://github.com/cern-sis/workflows/blob/990a8a2a3ee69688d85281c95d4de5c6829ee1ef/dags/common/pull_ftp.py#L159-L185
)
the template is to have double dash before the date

Reply:
I don't see the point make the query more complex than it should be. It is easier, cleaner and better approach to make the run_ids consistet

Reply:
and I see I did not replied  to your question  directly:
_it will affect all the ids from now on?_
Yes, all the tasks will have double underscore  before the date

Reply:
I still don't understand apologies. We already have a template with 1 underscore, right? Why changing? Do we set somewhere in airflow to add 2 underscores? From where these "autogenerated ids" are set?

Reply:
Can we have a call?

Reply:
No, we don't set the double underscore anywhere, it comes from Airflow itself.
Current implementation is wrong and not consistent, in my opinion, and I will try to show you why.

Let's say if the dag is triggered not by our written trigger by by schedular, the name of the run_id is:
scheduled__...
It comes by default from airflow, please the a look into  documentation : https://airflow.apache.org/docs/apache-airflow/1.10.14/_modules/airflow/models/dagrun.html#DagRun.run_id

I want to make our trigger created run_ids consistent with airflow run_ids. So, I think that adding one more underscore is a great and easy solution:
1) the name is consistent with Airflow naming
2) The SQL query for constructing logs names is simple. SQL query has to extract date from run_id, but having single underscore, it is harder, since we have two different variations:
Default ids, coming from airflow
a) scheduled__date, manual__date
and your custom ones
b) iop_date, oup_date, springer_date

While when we keeping it consistent with airlfow ids, we can have just one variation:
a)  scheduled__date,  manual__date, iop__date, oup__date, springer__date

In my personal opinion, the run_ids naming was not fully analysed, on the time it was written